pkg/generate/config/config.go

```diff
@@ -43,6 +44,12 @@ type OperationConfig struct {
 	// `resourceManager` struct that will set fields on a `resource` struct
 	// depending on the output of the operation.
 	SetOutputCustomMethodName string `json:"set_output_custom_method_name,omitempty"`
+	// Override for resource name in case of heuristic failure
+	// An example of this is correcting stutter when the resource logic doesn't properly determine the resource name
+	ResourceName  string            `json:"resource_name"`
+	// Override for operation type in case of heuristic failure
+	// An example of this is `Put...` or `Register...` API operations not being correctly classified as `Create` op type
+	OperationType operations.OpType `json:"operation_type"`
```

Comment:
> This PR also subpackaged `operations` in the model package to avoid cyclic dependency issues.

Hmm. OK. I think there's probably a way to structure the generatorConfig fields without having to create a new package for the operations codes.

You can just use a `string` data type here and do the translation to a `pkg/model.OpType` within the `pkg/model/crd.go` file.

Reply:
Switched back to the original package and using a `string` in the config instead.

To make it a bit more reusable, I instead added an `OpTypeFromString(s string)` method to `op.go`.